tools/spy-detector/Cargo.toml

```diff
@@ -0,0 +1,11 @@
+[package]
+name = "spy-detector"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+nng = "1.0.1"
+dashmap = "6.1.0"
```

Comment:
I see that you are using DashMap ("DashMap is an implementation of a concurrent associative array/hashmap in Rust."). You `std::thread::spawn` a thread to print the stats periodically, and the main thread collects the stats.   

I think you could print in the main thread by checking `STATS_INTERVAL` after a message arrived, which would reduce the overall complexity of the code and allows us to drop DashMap in favor of a `std::collection::HashMap`. I prefer to keep the dependency footprint minimal.

Reply:
Correct, I'll implement that asap!

Reply:
In that case, we can remove Arc and only use HashMap, since Arc provides immutable shared access to multiple threads.